src/views/token-minting/index.tsx

```
@@ -0,0 +1,265 @@
+  // In solana, an address for the wallet is handled by the spl program
+  // if it doesn't exists, it will be created.
+  // You can check the implementation of getAssociatedTokenAddress in:
+  // https://github.com/solana-labs/solana-program-library/blob/master/token/js/client/token.js#L2277
+  const associatedAddress = await Token.getAssociatedTokenAddress(
+    token.associatedProgramId,
+    token.programId,
+    token.publicKey,
+    address!
+  );
+  let associatedAccount: AccountInfo;
+  // This is the wallet version of {token.getOrCreateAssociatedAccountInfo}
+  try {
+    associatedAccount = await token.getAccountInfo(associatedAddress);
+  } catch (error) {
+    const associatedTokenAccountInstruction = Token.createAssociatedTokenAccountInstruction(
+      token.associatedProgramId,
+      token.programId,
+      token.publicKey,
+      associatedAddress,
+      address,
+      payer
+    );
+    const transactionId = await sendTransaction(
+      new Transaction().add(associatedTokenAccountInstruction),
+      connection
+    );
+    await connection.confirmTransaction(transactionId);
+    associatedAccount = await token.getAccountInfo(associatedAddress);
+  }
+  return associatedAccount;
+};
+
+export const TokenMinting: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [token, setToken] = useState<Token | null>(null);
+  const { connection } = useConnection();
+  const { publicKey, sendTransaction, signTransaction } = useWallet();
+
+  // Mint and send tokens form:
+  const [amount, setAmount] = useState(0);
+  const [destination, setDestination] = useState("");
+
+  const handleSubmitCreateMint: FormEventHandler<HTMLFormElement> = async (
+    e
+  ) => {
+    e.preventDefault();
+    try {
+      setIsLoading(true);
+
+      const mintAccount = Keypair.generate();
+      const token = new Token(
+        connection,
+        mintAccount.publicKey,
+        TOKEN_PROGRAM_ID,
+        {
+          publicKey: publicKey!,
```

This should probably check for the `publicKey` to be truthy and throw rather than use a non-null assertion for safety.